pycqed/analysis_v3/data_processing.py

```diff
@@ -577,318 +440,136 @@ def rotate_iq(data_dict, data_keys_in, data_keys_out, **params):
-    if len(data_keys_out) != len(data_to_proc_dict):
-        raise ValueError('data_keys_out and data_keys_in do not have '
+    data_to_proc_dict = help_func_mod.get_data_to_process(data_dict, keys_in)
+    keys_in = list(data_to_proc_dict)
+    if len(keys_out) != len(data_to_proc_dict):
+        raise ValueError('keys_out and keys_in do not have '
                          'the same length.')
-    if len(cp_list) != len(data_to_proc_dict):
-        raise ValueError('cal_points_list and data_keys_in do not have '
+
+    cp_list = help_func_mod.get_param('cal_points_list', data_dict, **params)
+    if cp_list is None:
+        cp = help_func_mod.get_param('cal_points', data_dict, **params)
+        if cp is None:
+            raise ValueError(
+                'Neither cal_points_list nor cal_points was found.')
+        else:
+            cp_list = [eval(cp)]*len(keys_in)
+    else:
+        cp_list = [eval(cp) for cp in cp_list]
+    if len(cp_list) != len(keys_in):
+        raise ValueError('cal_points_list and keys_in do not have '
                          'the same length.')
+    last_ge_pulses = help_func_mod.get_param('last_ge_pulses', data_dict,
+                                             default_value=[], **params)
+
+    mobjn = help_func_mod.get_param('meas_obj_name', data_dict,
+                                    raise_error=True, **params)
+    if mobjn not in cp.qb_names:
+        raise KeyError(f'{mobjn} not found in cal_points.')
 
     for j, keyi in enumerate(data_to_proc_dict):
         data = data_dict
-        all_keys = data_keys_out[j].split('.')
+        all_keys = keys_out[j].split('.')
         for i in range(len(all_keys)-1):
             if all_keys[i] not in data:
                 data[all_keys[i]] = OrderedDict()
             else:
                 data = data[all_keys[i]]
 
         cp = cp_list[j]
-        qbn = [k for k, v in meas_obj_value_names_map.items() if keyi == v][0]
         rotations = cp.get_rotations(last_ge_pulses=last_ge_pulses)
         ordered_cal_states = []
-        for ii in range(len(rotations[qbn])):
+        for ii in range(len(rotations[mobjn])):
             ordered_cal_states += \
-                [k for k, idx in rotations[qbn].items() if idx == ii]
+                [k for k, idx in rotations[mobjn].items() if idx == ii]
         rotated_data = \
             a_tools.rotate_and_normalize_data_1ch(
                 data=data_to_proc_dict[keyi],
                 cal_zero_points=None if len(ordered_cal_states) == 0 else
-                    cp.get_indices()[qbn][ordered_cal_states[0]],
+                    cp.get_indices()[mobjn][ordered_cal_states[0]],
                 cal_one_points=None if len(ordered_cal_states) == 0 else
-                    cp.get_indices()[qbn][ordered_cal_states[1]])
+                    cp.get_indices()[mobjn][ordered_cal_states[1]])
         data[all_keys[-1]] = rotated_data
     return data_dict
 
 
-## Helper functions ##
```

Comment:
I see there is now a module for helper functions. I think that is indeed better than having them inside another file. Yet it might be useful to have meaningful names for the relevant helper submodules, otherwise it might end like the analysis toolbox where there is plenty of things but you don't know what exactly or what the structure is (or there simply is no structure hahaha). This is not a comment specially relevant for the pull request but as I'm reading the code I'm just wondering what structure we want for this helper module.

Reply:
I agree. Let's keep this in mind as we develop. For now there are only a few functions there ;)